src/libsync/bulkpropagatorjob.h

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2021 (c) Matthieu Gallien <matthieu.gallien@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#ifndef BULKPROPAGATORJOB_H
+#define BULKPROPAGATORJOB_H
```

Comment:
> I would prefer pragma

why?

Reply:
1. Less code
2. Less prone to errors
3. Most compilers (especially the one we use) will be faster during the compilation because they don't open the file again if they encountered a `#pragma once` in the file before. Some compilers might also be able to do this with the include guards, but I'm unsure.

I think the only downside of using pragma is that it is not part of the standard. But I guess every compiler that we use and might use, supports it.